closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -38,7 +38,10 @@
 
 #define OPENSSL_VERSION_1_0 "OpenSSL 1.0."
 #define OPENSSL_VERSION_1_1 "OpenSSL 1.1."
-/* needed for OpenSSL 1.0.2 Thread handling routines */
+/* Per new OpenSSL naming convention starting from OpenSSL3, all major versions are ABI and API compatible. */
+#define OPENSSL_VERSION_3_0 "OpenSSL 3."
```

Comment:
This should be named `OPENSSL_VERSION_3_X` to reflect the value (it doesn't require the minor version to be `0`).

Reply:
Edited